packages/core/src/testInteg/sam.test.ts

```diff
@@ -46,262 +46,111 @@ const noDebugSessionInterval: number = 100
-        dependencyManager: 'npm',
-        vscodeMinimum: '1.78.0',
-    },
-    {
-        runtime: 'python3.10',
-        displayName: 'python 3.10 (ZIP)',
-        path: 'hello_world/app.py',
-        debugSessionType: 'python',
-        language: 'python',
-        dependencyManager: 'pip',
-        // https://github.com/microsoft/vscode-python/blob/main/package.json
-        vscodeMinimum: '1.77.0',
-    },
-    {
-        runtime: 'python3.11',
-        displayName: 'python 3.11 (ZIP)',
-        path: 'hello_world/app.py',
-        debugSessionType: 'python',
-        language: 'python',
-        dependencyManager: 'pip',
-        // https://github.com/microsoft/vscode-python/blob/main/package.json
-        vscodeMinimum: '1.78.0',
-    },
-    {
-        runtime: 'python3.12',
-        displayName: 'python 3.12 (ZIP)',
-        path: 'hello_world/app.py',
-        debugSessionType: 'python',
-        language: 'python',
-        dependencyManager: 'pip',
-        // https://github.com/microsoft/vscode-python/blob/main/package.json
-        vscodeMinimum: '1.78.0',
-    },
-    {
-        runtime: 'python3.13',
-        displayName: 'python 3.13 (ZIP)',
-        path: 'hello_world/app.py',
-        debugSessionType: 'python',
-        language: 'python',
-        dependencyManager: 'pip',
-        // https://github.com/microsoft/vscode-python/blob/main/package.json
-        vscodeMinimum: '1.78.0',
-    },
-    {
-        runtime: 'dotnet6',
-        displayName: 'dotnet6 (ZIP)',
-        path: 'src/HelloWorld/Function.cs',
-        debugSessionType: 'coreclr',
-        language: 'csharp',
-        dependencyManager: 'cli-package',
-        vscodeMinimum: '1.80.0',
-    },
-    {
-        runtime: 'java8.al2',
-        displayName: 'java8.al2 (Maven ZIP)',
-        path: 'HelloWorldFunction/src/main/java/helloworld/App.java',
-        debugSessionType: 'java',
-        language: 'java',
-        dependencyManager: 'maven',
-        vscodeMinimum: '1.50.0',
-    },
-    {
-        runtime: 'java11',
-        displayName: 'java11 (Gradle ZIP)',
-        path: 'HelloWorldFunction/src/main/java/helloworld/App.java',
-        debugSessionType: 'java',
-        language: 'java',
-        dependencyManager: 'gradle',
-        vscodeMinimum: '1.50.0',
-    },
-    {
-        runtime: 'java17',
-        displayName: 'java11 (Gradle ZIP)',
-        path: 'HelloWorldFunction/src/main/java/helloworld/App.java',
-        debugSessionType: 'java',
-        language: 'java',
-        dependencyManager: 'gradle',
-        vscodeMinimum: '1.50.0',
-    },
-    // {
-    //     runtime: 'go1.x',
-    //     displayName: 'go1.x (ZIP)',
-    //     path: 'hello-world/main.go',
-    //     debugSessionType: 'delve',
-    //     language: 'go',
-    //     dependencyManager: 'mod',
-    //     // https://github.com/golang/vscode-go/blob/master/package.json
-    //     vscodeMinimum: '1.67.0',
-    // },
-
+    generateScenario('nodejs', '18.x'),
```

Comment:
A benefit of the old code is that it was "flat" data that did not require inspecting any logic. Copy-pasting can be very costly for logic, but for data, especially data contained in one place (this file) it can be a useful tradeoff.

However, your new code is certainly easier to read at a glance.